crl-release-25.4: TestIngestLoadRand failed (leaked goroutine)

The nightly linux-stress test run failed on b9a9fc99139db79a4121f895f1c3cf5322704ed3 (go1.23). Please review the run [here](https://github.com/cockroachdb/pebble/actions/runs/24239180652).

Jira issue: PEBBLE-1383

```
--- FAIL: TestIngestLoadRand (5.06s)
    ingest_test.go:265: 
    ingest_test.go:267: Leaked goroutine: 124038
        
        
        all stacks: 
        
        goroutine 124079 [running]:
```

hmm. maybe something to do with my backport; i'll check it out

also leaked goroutine in:
```
--- FAIL: TestIngestLoadInvalid (5.05s)
    ingest_test.go:284: Leaked goroutine: 121672
        
        
        all stacks: 
        
        goroutine 121653 [running]:
        github.com/cockroachdb/crlib/testutils/leaktest.getStacks()
        	C:/Users/runneradmin/go/pkg/mod/github.com/cockroachdb/crlib@v0.0.0-20250916151006-1094cb39adac/testutils/leaktest/leaktest.go:186 +0x49
        github.com/cockroachdb/crlib/testutils/leaktest.AfterTest.func2()
        	C:/Users/runneradmin/go/pkg/mod/github.com/cockroachdb/crlib@v0.0.0-20250916151006-1094cb39adac/testutils/leaktest/leaktest.go:151 +0x19c
        github.com/cockroachdb/pebble.TestIngestLoadInvalid(0xc0000d4b60)
        	D:/a/pebble/pebble/ingest_test.go:284 +0x2ef
        testing.tRunner(0xc0000d4b60, 0x7ff629552c68)
        	C:/hostedtoolcache/windows/go/1.23.12/x64/src/testing/testing.go:1690 +0xcb
        created by testing.(*T).Run in goroutine 1
        	C:/hostedtoolcache/windows/go/1.23.12/x64/src/testing/testing.go:1743 +0x377
```

--- a/compaction_test.go
+++ b/compaction_test.go
@@ -3356,6 +3356,7 @@ func TestTombstoneDensityCompactionMoveOptimization(t *testing.T) {
 		return NewConcurrencyLimitSchedulerWithNoPeriodicGrantingForTest()
 	}
 	opts.WithFSDefaults()
+	defer opts.private.fsCloser.Close()
 
 	// Create a file with high tombstone density.
 	meta := &manifest.TableMetadata{
@@ -3458,6 +3459,7 @@ func TestTombstoneDensityCompactionMoveOptimization_NoMoveWithOverlap(t *testing
 		return NewConcurrencyLimitSchedulerWithNoPeriodicGrantingForTest()
 	}
 	opts.WithFSDefaults()
+	defer opts.private.fsCloser.Close()
 
 	// Create a file with high tombstone density in L4.
 	metaL4 := &manifest.TableMetadata{
@@ -3542,6 +3544,7 @@ func TestTombstoneDensityCompactionMoveOptimization_GrandparentOverlapTooLarge(t
 		return NewConcurrencyLimitSchedulerWithNoPeriodicGrantingForTest()
 	}
 	opts.WithFSDefaults()
+	defer opts.private.fsCloser.Close()
 
 	// File in L4 with high tombstone density.
 	metaL4 := &manifest.TableMetadata{
@@ -3610,6 +3613,7 @@ func TestTombstoneDensityCompactionMoveOptimization_BelowDensityThreshold(t *tes
 		return NewConcurrencyLimitSchedulerWithNoPeriodicGrantingForTest()
 	}
 	opts.WithFSDefaults()
+	defer opts.private.fsCloser.Close()
 
 	meta := &manifest.TableMetadata{
 		TableNum: 1,
@@ -3662,6 +3666,7 @@ func TestTombstoneDensityCompactionMoveOptimization_InvalidStats(t *testing.T) {
 		return NewConcurrencyLimitSchedulerWithNoPeriodicGrantingForTest()
 	}
 	opts.WithFSDefaults()
+	defer opts.private.fsCloser.Close()
 
 	meta := &manifest.TableMetadata{
 		TableNum: 1,

--- a/ingest_test.go
+++ b/ingest_test.go
@@ -115,6 +115,7 @@ func TestIngestLoad(t *testing.T) {
 			}
 			var bv blobtest.Values
 			w := sstable.NewRawWriter(objstorageprovider.NewFileWritable(f), writerOpts)
+			defer w.Close()
 			for data := range crstrings.LinesSeq(td.Input) {
 				if strings.HasPrefix(data, "Span: ") {
 					data = strings.TrimPrefix(data, "Span: ")
@@ -154,6 +155,7 @@ func TestIngestLoad(t *testing.T) {
 				FS:         mem,
 			}
 			opts.WithFSDefaults()
+			defer opts.private.fsCloser.Close()
 			getNextFileNum := func() base.DiskFileNum { return 1 }
 			localFiles := LocalSSTables([]LocalSST{{Path: "ext"}})
 			lr, err := ingestLoad(context.Background(), opts, dbVersion, localFiles, nil, nil, nil, nil, getNextFileNum)
@@ -266,6 +268,7 @@ func TestIngestLoadRand(t *testing.T) {
 		FS:       mem,
 	}
 	opts.WithFSDefaults()
+	defer opts.private.fsCloser.Close()
 	opts.EnsureDefaults()
 	lr, err := ingestLoad(context.Background(), opts, version, paths, nil, nil, nil, nil, fileNumAllocator.nextFileNum)
 	require.NoError(t, err)
@@ -748,6 +751,7 @@ func TestIngestLoadInvalid(t *testing.T) {
 		FS:       mem,
 	}
 	opts.WithFSDefaults()
+	defer opts.private.fsCloser.Close()
 	getNextFileNum := func() base.DiskFileNum { return 1 }
 	localFiles := LocalSSTables([]LocalSST{{Path: "invalid"}})
 	if _, err := ingestLoad(context.Background(), opts, internalFormatNewest, localFiles, nil, nil, nil, nil, getNextFileNum); err == nil {
@@ -1999,6 +2003,7 @@ func TestIngestMemtableOverlaps(t *testing.T) {
 					}
 					opts.EnsureDefaults()
 					opts.WithFSDefaults()
+					defer opts.private.fsCloser.Close()
 					if len(d.CmdArgs) > 1 {
 						return fmt.Sprintf("%s expects at most 1 argument", d.Cmd)
 					}

